### PR TITLE
MODQM-262 - MARC field protection doesn't verify the subfield value

### PR DESCRIPTION
**Pull request by oliinyko**
## Purpose
Fix issue with MARC field protection subfield verification

JIRA: https://issues.folio.org/browse/MODQM-262

## Approach
- Fix pattern for subfield verification
- Add tests

### Changes checklist
- [ ] API paths, methods, request or response bodies changed, added, or removed
- [ ] Database schema changes
- [ ] Interface versions changes
- [ ] Interface dependencies added, or removed
- [ ] Permissions changed, added, or removed

### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

### Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

File: src/main/java/org/folio/qm/service/impl/FieldProtectionSetterServiceImpl.java [hunk 1]
```diff
@@ -93,14 +93,17 @@ private boolean isAnyIndicator2InSettingOrIndicator2Match(MarcFieldProtectionSet
 
   private boolean isAnySubFieldInSettingOrSubFieldMatch(MarcFieldProtectionSetting setting, FieldItem field) {
     return setting.getSubfield().equals(ANY_STRING)
-      || !Pattern.compile("[$]" + setting.getSubfield()).matcher(field.getContent().toString()).matches();
+      || Pattern.compile("[$]" + setting.getSubfield() + "\\b").matcher(field.getContent().toString()).find();
```
Comment (psmagin) [inline src/main/java/org/folio/qm/service/impl/FieldProtectionSetterServiceImpl.java:96]: Will it work if a setting has protected $b and field content is: $atest1$btest2 ?

Comment (oliinyko) [reply src/main/java/org/folio/qm/service/impl/FieldProtectionSetterServiceImpl.java:96]: no it won't. Is it a valid scenario to set subfield values without whitespaces?

Comment (psmagin) [reply src/main/java/org/folio/qm/service/impl/FieldProtectionSetterServiceImpl.java]: @oliinyko yes, it's possible:
image

Comment (oliinyko) [reply src/main/java/org/folio/qm/service/impl/FieldProtectionSetterServiceImpl.java]: ok, I've removed word boundary from pattern

**Review by viacheslavkol — APPROVED** [review]

**Review by psmagin — APPROVED** [review]

**Pull request merged**